### ISSUE TITLE
[Accessibility Audit] Code Intel: Read file content

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#34448]
### Steps to replicate journey

Taken from the [user journey list document](https://docs.google.com/document/d/1kA6aVOAgID_uPm-d6uEC1DG6WHjJ8fPT0UAXQcML_KQ/edit?usp=sharing). Use this for further context.

Note: This is primarily aimed to assess the _consumption_ of content. As in, it should be possible to read the content of a file without any issues, irrelevant of if a user has a screen reader, magnifier, etc.

- Navigate to a file in Sourcegraph
- Read file contents

### How to audit

Follow the instructions here: [Auditing a user journey](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#auditing-a-user-journey)

**Note:** We are planning on arranging thorough third-party audit in the future, so our focus here is just to find the _majority_ of accessibility issues. Don't worry if you feel you haven't found 100% of potential issues, it is better to focus on the core essentials to complete the journey rather than spending lots of time going through every possible problem.

### How to raise discovered problems

Follow the instructions here: [Raising an accessibility bug](https://docs.sourcegraph.com/dev/background-information/web/accessibility/how-to-audit#raising-a-bug)

### Any other questions?

See the [audit FAQ](https://docs.google.com/document/d/1nCi7TvYRpj9eG0_QtgqK1c4L28EiJKVuxyuvg1lAGgo/edit#) - add your question if it isn't there already!

**Comment by umpox** [comment]
cc: @varungandhi-src. I know this is mostly covered by other journeys (e.g. navigate to references) but I think it's important that we have a dedicated user journey just for reading a file. LMK if there's any issues

**Comment by varungandhi-src** [comment]
(Moved from https://github.com/sourcegraph/sourcegraph/issues/33509)

## Blob view (5)

- [X] It is difficult to navigate to the code blob itself (no header structure) when using a screen reader. https://github.com/sourcegraph/sourcegraph/issues/35072
  * Potential fix: Add header for code view.
  * Other follow-ups: Add headers for other main parts of the UI, such as the products layer, the search bar layer, the navigation and metadata layer (the filepath breadcrumbs + brain dot, github link, etc), the sidebar, and the reference panel.
- [X] The code is formatted as a table, which feels incorrect (doesn't match editors like GoLand). It seems like we can navigate down the correct rows without having to read line numbers or gutters each time. Not sure if there was something else notable to mention here (will defer to @varungandhi-src's opinions on this as he was driving). https://github.com/sourcegraph/sourcegraph/issues/36521
- [X] Clicking on on a line number trigger highlights + a URL change (unlike a mouse click). https://github.com/sourcegraph/sourcegraph/issues/36526
- [X] Needlessly clickable elements: all table entries are marked clickable, but clicking doesn't do anything for a lot of them. For example, `import (`, `)` at the end of imports etc. https://github.com/sourcegraph/sourcegraph/issues/35267
- [X] The code view seems to have a table inside a group, making it tricky to go to the first line quickly. You need to Tab to the group, then Navigate to Prev Line (Control + Option + ←) then Navigate to Next Line (Control + Option + →). Trying Navigate to Next Line directly from the group takes one to the next item (e.g. a git blame URL). https://github.com/sourcegraph/sourcegraph/issues/36531

**Comment by varungandhi-src** [comment]
Flagging some issues I just noticed:

- [ ] It doesn't seem possible to navigate to the file paths at the top of the code view using keyboard navigation.
- [ ] The file paths are not visible for narrow screens. but the copy path to clipboard button is next to the repo drop down.
- [ ] It doesn't seem possible to select multiple lines on a mobile screen.

**Comment by varungandhi-src** [comment]
Closing this; I think we've documented all the issues we ran into.
